src/state/slices/config_slice.ts

```diff
@@ -65,3 +78,58 @@ export const createConfigSlice: StateCreator<ConfigSlice, [], [], ConfigSlice> =
   setOutlineColor: (outlineColor) => set({ outlineColor }),
   setOpenTab: (openTab) => set({ openTab }),
 });
+
+export const serializeConfigSlice = (slice: ConfigSlice): SerializedStoreData => {
+  const ret: SerializedStoreData = {};
+  ret[UrlParam.SHOW_PATH] = encodeBoolean(slice.showTrackPath);
+  ret[UrlParam.SHOW_SCALEBAR] = encodeBoolean(slice.showScaleBar);
+  ret[UrlParam.SHOW_TIMESTAMP] = encodeBoolean(slice.showTimestamp);
+  // Export settings are currently not serialized.
+  ret[UrlParam.FILTERED_COLOR] = encodeColor(slice.outOfRangeDrawSettings.color);
+  ret[UrlParam.FILTERED_MODE] = slice.outOfRangeDrawSettings.mode.toString();
+  ret[UrlParam.OUTLIER_COLOR] = encodeColor(slice.outlierDrawSettings.color);
+  ret[UrlParam.OUTLIER_MODE] = slice.outlierDrawSettings.mode.toString();
+  ret[UrlParam.OUTLINE_COLOR] = encodeColor(slice.outlineColor);
+
+  ret[UrlParam.OPEN_TAB] = slice.openTab;
+  return ret;
+};
+
+export const loadConfigSliceFromParams = (slice: ConfigSlice, params: URLSearchParams): void => {
+  const showPathParam = decodeBoolean(params.get(UrlParam.SHOW_PATH));
+  if (showPathParam !== undefined) {
+    slice.setShowTrackPath(showPathParam);
+  }
+  const showScaleBarParam = decodeBoolean(params.get(UrlParam.SHOW_SCALEBAR));
+  if (showScaleBarParam !== undefined) {
+    slice.setShowScaleBar(showScaleBarParam);
+  }
+  const showTimestampParam = decodeBoolean(params.get(UrlParam.SHOW_TIMESTAMP));
+  if (showTimestampParam !== undefined) {
+    slice.setShowTimestamp(showTimestampParam);
+  }
```

Comment:
do these not have defaults?

this code is repetitive, you could do 
`setBooleanParam(param, callback)` and move the check to see if it's undefined within the component. Or you could do `setBooleanParam(param, callback, default)`

Reply:
They do have default values, which are defined above at initialization!
 I do want to avoid setting the flags when they're undefined but I'll also try shortening these using a method like you suggested.